cmd/juju/interact/pollster.go

```diff
@@ -393,7 +394,11 @@ func (p *Pollster) queryOneSchema(schema *jsonschema.Schema) (interface{}, error
 		// anything
 		a, err = p.Enter(schema.Singular)
 	case jsonschema.FormatURI:
```

Comment:
So, part of it is that there's not really supposed to be custom per-cloud code.  The whole idea of using jsonschema to generate the interactive UI is so that we don't write custom code.  VerifyUrls already breaks this, unfortunately, but we can probably fix that later.

Second, verifyURLs works because it's specific - it knows that it's only getting passed values for endpoints, so it can just verify the endpoint.  However, if we genericized it, then we'd need to pass the jsonschema with the value to verify it, so we can tell what it is this value represents... but you really would need to pass the whole schema, not just a particular sub-schema, because you wouldn't be able to tell if this is like the name of an endpoint of a cloud, or just a name of a cloud, etc.  .... and at that point, you're basically just writing custom verification code for the whole cloud config, in which case, the jsonschema is not actually useful (btw, that's actually what I wanted to do in the first place, but was shot down).